python/sglang/srt/entrypoints/engine.py

```diff
@@ -301,14 +302,41 @@ def init_weights_update_group(
 
     def update_weights_from_distributed(self, name: str, dtype, shape):
         """Update weights from distributed source."""
-        obj = UpdateWeightsFromDistributedReqInput(
-            name=name,
-            dtype=dtype,
-            shape=shape,
+        return self.batch_update_weights_from_distributed(
+            [{"name": name, "dtype": dtype, "shape": shape}]
         )
+
+    def batch_update_weights_from_distributed(self, parameters: List[Dict]):
+        """
+        Update multiple weights from distributed source in a single batch operation.
+
+        Args:
+            parameters: A list of dictionaries, where each dictionary contains:
+                - name: The name of the parameter to update
+                - dtype: The data type of the parameter (can be a string or torch.dtype)
+                - shape: The shape of the parameter
+
+        Returns:
+            A tuple of (success, message)
+        """
+        # Convert torch.dtype objects to strings if needed
+        processed_parameters = []
+        for param in parameters:
+            if not isinstance(param["dtype"], str):
+                dtype_str = str(param["dtype"]).split(".")[-1]
+            else:
+                dtype_str = param["dtype"]
+
+            processed_parameters.append(
+                UpdateWeightsFromDistributedReqInput(
+                    name=param["name"], dtype=dtype_str, shape=param["shape"]
+                )
+            )
```

Comment:
nit: what about

```
processed_parameters = [UpdateWeightsFromDistributedReqInput.init_new(**param) for param in parameters]
```

where we define `UpdateWeightsFromDistributedReqInput.init_new` to contain type conversion

Reply:
I actually think we should not have a `init_new` in the dataclass for `UpdateWeightsFromDistributedReqInput` since no other dataclasses in `io_struct` has such method. I guess we can put the dtype conversion inline.